main.tf

```diff
@@ -1,13 +1,99 @@
-module "vm" {
-  source               = "./modules/vm"
-  network_interface_id = var.vpc_id
-  graphdb_subnets      = var.graphdb_subnets
-  instance_type        = var.instance_type
-  lb_subnets           = var.lb_subnets
+provider "azurerm" {
+  features {
+    managed_disk {
+      expand_without_downtime = true
+    }
+    resource_group {
+      prevent_deletion_if_contains_resources = true
+    }
+  }
+}
+
+locals {
+  tags = merge({
+    # Used to easily track all resource managed by Terraform
+    deployment = var.resource_name_prefix
+    source     = "Terraform"
+  }, var.tags)
+}
+
+# ------------------------------------------------------------
+
+# TODO: To be moved in another module/example module + configurations
+
+resource "azurerm_resource_group" "graphdb" {
+  name     = var.resource_name_prefix
+  location = var.location
+  tags     = local.tags
+}
+
+resource "azurerm_virtual_network" "graphdb" {
+  name                = var.resource_name_prefix
+  resource_group_name = azurerm_resource_group.graphdb.name
+  location            = azurerm_resource_group.graphdb.location
+  address_space       = ["10.0.0.0/16"]
+  tags                = local.tags
+}
+
+resource "azurerm_subnet" "graphdb-private" {
+  name                 = "${var.resource_name_prefix}-private"
+  resource_group_name  = azurerm_resource_group.graphdb.name
+  virtual_network_name = azurerm_virtual_network.graphdb.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+# ------------------------------------------------------------
+
+# TODO: Could go into another module
+
+# TODO: Config + how to .. refer other account group?
+data "azurerm_resource_group" "image" {
+  name = "Packer-RG"
+}
+
+# TODO: Support for multiple architectures
```

Comment:
TODO as well - when we release the image officially, we can move to [platform_image](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/platform_image).